src/general.rs

```diff
@@ -76,7 +76,9 @@ pub fn initialize(root_path: &str, meta: EditorMeta, ctx: &mut Context) {
                 }),
                 workspace_folders: Some(false),
                 configuration: Some(true),
-                semantic_tokens: None,
+                semantic_tokens: Some(SemanticTokensWorkspaceClientCapabilities {
+                    refresh_support: Some(true),
```

Comment:
Looks like this makes rust-analyzer send lots of `workspace/semanticTokens/refresh` messages, even if we never actively ask for them elsewhere.
Same thing happens with `$/progress`.

It's probably not a problem but it felt a bit weird. I guess that's just because I'm looking at the logs.
So that amount of traffic should be perfectly fine, especially since it's initiated by the server (unlike our NormalIdle hooks).
(Related: I think we should turn off requesting code actions by default (#549).)

I wonder if there is a nice way to detect if the user wants to use semantic tokens / progress notifications?

Going forward, I think we should allow to turn off features without recompiling (if only for debugging).
Here we are talking about initialization options.

So I think we should allow to configure this via `kak-lsp.toml` (like `snippet_support`).
What about:

```toml
[InitializeParams.capabilities.workspace.semanticTokens]
refresh_support = false
```

and also, dynamically:

```kak
set-option global lsp_config %{
	[InitializeParams.capabilities.workspace.semanticTokens]
	refresh_support = false
}
```

I guess the dynamic variant is not so important for initialization, since you can always restart the language server. But at least for rust-analyzer that can take a while.

Reply:
I don't think this is necessarily an issue between kak-lsp's server and the LSP server - it might be sufficient to add some configuration to whether kak-lsp should *relay* those events to the kakoune instance.